### PR TITLE
fix(addon-commerce): `InputCardGroup` fix Safari autofill

### LINKED ISSUE #9188: 🐞 - InputCardGroup wrong autofill behavior in Safari
### Code Example URL

https://taiga-ui.tcsbank.ru/components/input-card-group

### Description

When use autofill to paste card data (num, expire date), focus remains on card number input, don't switch to CVC field. Also, if clear input after autofill, styles (background and placeholder color) remains as it was during autofill:
Снимок экрана 2024-09-25 в 15 17 49

### Taiga UI version

4.8.1

### Browser Used

Safari

### OS Used

macOS

### PR DESCRIPTION
**Pull request by waterplea**
Close #9188

### CODE REVIEW

File: projects/addon-commerce/components/input-card-group/input-card-group.component.ts [hunk 1]
```diff
@@ -307,7 +313,8 @@ export class TuiInputCardGroup
         this.updateBin(bin);
 
         if (this.cardValidator(this.card) && !value()?.expire && this.inputExpire) {
-            this.focusExpire();
+            // Safari autofill focus jerk workaround
+            setTimeout(() => this.focusExpire());
```
Comment (splincode) [inline projects/addon-commerce/components/input-card-group/input-card-group.component.ts:317]: please add runOutsideAngular

File: projects/core/components/textfield/textfield.directive.ts [hunk 2]
```diff
@@ -67,8 +67,13 @@ export class TuiTextfieldBase<T> implements OnChanges {
     }
 
     public setValue(value: T | null): void {
-        this.el.value = value == null ? '' : this.textfield.stringify(value);
-        this.el.dispatchEvent(new Event('input', {bubbles: true}));
+        this.el.focus();
+        this.el.select();
+        this.el.ownerDocument.execCommand(
```
Comment (splincode) [inline projects/core/components/textfield/textfield.directive.ts:72]: what about `execCommand is deprecated`?
image

**Review by splincode — APPROVED** [review]

File: projects/addon-commerce/components/input-card-group/input-card-group.component.ts (hunk 1 above)
Comment (waterplea) [reply projects/addon-commerce/components/input-card-group/input-card-group.component.ts:317]: Doesn't matter, focus will trigger change detection it anyway

File: projects/core/components/textfield/textfield.directive.ts (hunk 2 above)
Comment (waterplea) [reply projects/core/components/textfield/textfield.directive.ts:72]: It's not going away anytime soon, no other way to kick Safari to work properly

File: projects/addon-commerce/components/input-card-group/input-card-group.component.ts [hunk 3]
```diff
@@ -309,12 +310,13 @@ export class TuiInputCardGroup
             return;
         }
 
+        clearInterval(this.timeout);
```
Comment (splincode) [inline projects/addon-commerce/components/input-card-group/input-card-group.component.ts:313]: `ngOnDestroy` also?

File: projects/addon-commerce/components/input-card-group/input-card-group.component.ts [hunk 4]
```diff
@@ -309,12 +310,13 @@ export class TuiInputCardGroup
...
         this.updateProperty(parsed, 'card');
         this.updateBin(bin);
 
         if (this.cardValidator(this.card) && !value()?.expire && this.inputExpire) {
             // Safari autofill focus jerk workaround
-            setTimeout(() => this.focusExpire());
+            this.timeout = Number(setTimeout(() => this.focusExpire(), 100));
```
Comment (splincode) [inline projects/addon-commerce/components/input-card-group/input-card-group.component.ts:319]: maybe

```ts
timer(100)
.pipe(takeUntilDestroy(this.destroyRef))
.subscribe(() => this.focusExpire())
```

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**